framework/src/transfers/MultiAppFieldTransferInterface.C

```diff
@@ -0,0 +1,396 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "MultiAppFieldTransferInterface.h"
+#include "MooseTypes.h"
+#include "FEProblem.h"
+#include "MultiApp.h"
+#include "MooseMesh.h"
+#include "UserObject.h"
+#include "NearestPointIntegralVariablePostprocessor.h"
+
+template <>
+InputParameters
+validParams<MultiAppFieldTransferInterface>()
+{
+  InputParameters params = validParams<MultiAppTransfer>();
+  params.addRequiredParam<std::vector<AuxVariableName>>(
+      "variable", "The auxiliary variable to store the transferred values in.");
+  params.addRequiredParam<std::vector<VariableName>>("source_variable",
+                                                     "The variable to transfer from.");
+  params.addParam<bool>("preserve_transfer",
+                        false,
+                        "Whether or not to conserve the transfered field, "
+                        " if true, the transfered variables will be adjusted "
+                        "according to the pps value");
+
+  std::vector<PostprocessorName> from_postprocessor = {"from_postprocessor"};
+  params.addParam<std::vector<PostprocessorName>>(
+      "from_postprocessor_to_be_preserved",
+      from_postprocessor,
+      "The name of the Postprocessor in the from-app to evaluate an adjusting factor.");
+
+  std::vector<PostprocessorName> to_postprocessor = {"to_postprocessor"};
+  params.addParam<std::vector<PostprocessorName>>(
+      "to_postprocessor_to_be_preserved",
+      to_postprocessor,
+      "The name of the Postprocessor in the to-app to evaluate an adjusting factor.");
+  return params;
+}
+
+MultiAppFieldTransferInterface::MultiAppFieldTransferInterface(const InputParameters & parameters)
```

Comment:
Should we just name this as `MultiAppFieldTransfer` since it is in the inheritance chain but not a separate interface?